presto-native-execution/src/test/java/com/facebook/presto/nativeworker/ContainerQueryRunnerUtils.java

```diff
@@ -0,0 +1,332 @@
+        }
+    }
+
+    public static void createPropertiesFile(String filePath, Properties properties)
+    {
+        try {
+            File file = new File(filePath);
+
+            File parentDir = file.getParentFile();
+            if (!parentDir.exists()) {
+                parentDir.mkdirs();
+            }
+
+            if (file.exists()) {
+                throw new IOException("File exists: " + filePath);
+            }
+
+            try (FileWriter writer = new FileWriter(file)) {
+                for (String key : properties.stringPropertyNames()) {
+                    writer.write(key + "=" + properties.getProperty(key) + "\n");
+                }
+                logger.info(file.getName() + " created successfully with content : " + properties);
+            }
+        }
+        catch (IOException io) {
+            io.printStackTrace();
+        }
+    }
+
+    public static void createScriptFile(String filePath, String scriptContent)
+    {
+        try {
+            File file = new File(filePath);
+
+            File parentDir = file.getParentFile();
+            if (!parentDir.exists()) {
+                parentDir.mkdirs();
+            }
+
+            if (file.exists()) {
+                throw new IOException("File exists : " + filePath);
+            }
+
+            try (OutputStream output = new FileOutputStream(file)) {
+                output.write(scriptContent.getBytes());
+            }
+
+            file.setExecutable(true);
+            logger.info(file.getName() + " created successfully with content :" + scriptContent);
+        }
+        catch (IOException io) {
+            io.printStackTrace();
+        }
+    }
+
+    public static MaterializedResult toMaterializedResult(String csvData)
+    {
+        try {
+            List<Type> columnTypes = new ArrayList<>();
+
+            // Parse CSV data using OpenCSV
+            CSVReader reader = new CSVReader(new StringReader(csvData));
+            List<String[]> records = reader.readAll();
+
+            // Collect all rows as lists of strings
+            ImmutableList.Builder<List<String>> allRowsBuilder = ImmutableList.builder();
+            for (String[] record : records) {
+                allRowsBuilder.add(ImmutableList.copyOf(record));
+            }
+            ImmutableList<List<String>> allRows = allRowsBuilder.build();
+
+            // Infer column types based on the maximum columns found
+            int maxColumns = allRows.stream().mapToInt(List::size).max().orElse(0);
+            for (int i = 0; i < maxColumns; i++) {
+                final int columnIndex = i;
+                columnTypes.add(inferType(allRows.stream()
+                        .map(row -> columnIndex < row.size() ? row.get(columnIndex) : "")
+                        .collect(Collectors.toList())));
+            }
+
+            // Convert all rows to MaterializedRow
+            ImmutableList.Builder<MaterializedRow> rowsBuilder = ImmutableList.builder();
```

Comment:
In local variables in tests we don't gain much from immutability. I'd prefer the simplicity of plain array lists. Up to you.

Reply:
Did the changes as per the [comment](https://github.com/prestodb/presto/pull/23094#discussion_r1688126043)